rca/shortcourses/tests/test_access_planit.py

```diff
@@ -0,0 +1,217 @@
+        # logging.disable(logging.CRITICAL)  # Don't log exceptions for tests... is this bad?
+        mock_get.side_effect = Timeout
+        data = AccessPlanitXML(course_id=1)
+        xml_data = data.get_data()
+        self.assertEqual(xml_data, [])
+
+    def test_management_command_fetch_data(self):
+        """ Test that xml with no dates goes in the cache as [] """
+        for i in range(5):
+            ShortCoursePage.objects.create(
+                title=f"Short course {i}",
+                path=str(i),
+                depth="001",
+                access_planit_course_id=i,
+            )
+        args = []
+        opts = {}
+        call_command("fetch_access_planit_data", *args, **opts)
+        for i in range(5):
+            cache_key = f"short_course_{i}"
+            self.assertEqual(cache.get(cache_key), [])
+
+    @mock.patch(
+        "rca.shortcourses.access_planit.AccessPlanitXML.fetch_data_from_xml",
+        side_effect=mocked_fetch_data_from_xml,
+    )
+    def test_management_command_fetch_with_example_data(
+        self, mocked_fetch_data_from_xml
+    ):
+        """ Test that the example xml goes in the cache as expected """
+        for i in range(5):
+            ShortCoursePage.objects.create(
+                title=f"Short course {i}",
+                path=str(i),
+                depth="001",
+                access_planit_course_id=i,
+            )
+        args = []
+        opts = {}
+        call_command("fetch_access_planit_data", *args, **opts)
+        for i in range(5):
+            cache_key = f"short_course_{i}"
+            self.assertEqual(cache.get(cache_key), self.expected_data)
+
+    @mock.patch("rca.shortcourses.access_planit.requests.get")
+    def test_page_renders_with_timeout(self, mock_get):
+        logging.disable(
+            logging.CRITICAL
+        )  # Silences the timeout raised from mock when running the test
+        """ If there is a timeout for the xml request when the page loads,
+        ensure the page still renders with the empty data"""
+        home_page = HomePage.objects.first()
+        mock_get.side_effect = Timeout
+        short_course_page = ShortCoursePage(
+            title="Short course title", slug="short-course", access_planit_course_id="1"
+        )
+        home_page.add_child(instance=short_course_page)
+        response = self.client.get("/short-course/")
+        self.assertTemplateUsed(
+            response, "patterns/pages/shortcourses/short_course.html"
+        )
+        self.assertContains(response, "Short course title")
+        self.assertEqual(
+            response.render().status_code, 200
+        )  # will render 404 if there is an failure
+
+    @mock.patch(
+        "rca.shortcourses.access_planit.AccessPlanitXML.fetch_data_from_xml",
+        side_effect=mocked_fetch_data_from_xml,
+    )
+    def test_page_renders_good_xml(self, mocked_fetch_data_from_xml):
```

Comment:
Good page integration test 👍 

(Tests that use the Django test client to `.get()` can be viewed as integration tests)

Reply:
:tada: